pkg/controller/common/featuregates.go

```diff
@@ -0,0 +1,47 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+	"k8s.io/klog/v2"
+)
+
+func WaitForFeatureGatesReady(ctx context.Context, featureGateAccess featuregates.FeatureGateAccess) error {
+	timeout := time.After(1 * time.Minute)
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-timeout:
+			return fmt.Errorf("timed out waiting for FeatureGates to be ready")
+		default:
+			features, err := featureGateAccess.CurrentFeatureGates()
+			if err == nil {
+				enabled, disabled := GetEnabledDisabledFeatures(features)
+				klog.Infof("FeatureGates initialized: enabled=%v, disabled=%v", enabled, disabled)
+				return nil
+			}
+			klog.Infof("Waiting for FeatureGates to be ready...")
```

Comment:
Interesting, if we look at the bootstrap test logs now:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_machine-config-operator/4439/pull-ci-openshift-machine-config-operator-master-bootstrap-unit/1806726663283675136/artifacts/test/build-log.txt

It always takes exactly 1 retry (1 second) for the featuregates to become ready. Is that... normal? I guess the bootstrap tests should be relatively fast.

@QiWang19 does this match your expectations?

Reply:
The result from the test log looks right. The featuregates are available before the controller starts.